### ISSUE TITLE
Chromium not found in actor-node-playwright-chrome

### ISSUE THREAD

**Issue by vanekj** [apify/apify-actor-docker#87]
Hello!

I ran into an issue while configuring my crawler Dockerfile with `apify/actor-node-playwright-chrome` image.

When I build and run my Dockerfile, I get this error and it stops.
```
ERROR PlaywrightCrawler: Request failed and reached maximum retries. browserType.launchPersistentContext:
Executable doesn't exist at /home/myuser/pw-browsers/chromium-1028/chrome-linux/chrome
╔═════════════════════════════════════════════════════════════════════════╗
║ Looks like Playwright Test or Playwright was just installed or updated. ║
║ Please run the following command to download new browsers:              ║
║                                                                         ║
║     npx playwright install                                              ║
║                                                                         ║
║ <3 Playwright Team                                                      ║
╚═════════════════════════════════════════════════════════════════════════╝
```

Could you please help me, what can I change to make it work?

Thank you! 🙏🏻 

---

**Dockerfile**
```
FROM apify/actor-node-playwright-chrome:18

COPY --chown=myuser:myuser package*.json ./

RUN npm --quiet set progress=false
RUN npm ci --only=production

COPY --chown=myuser:myuser . ./

CMD ["node", "src/main.js"]
```

**.dockerignore**
```
**/.classpath
**/.dockerignore
**/.env
**/.git
**/.gitignore
**/.project
**/.settings
**/.toolstarget
**/.vs
**/.vscode
**/*.*proj.user
**/*.dbmdl
**/*.jfm
**/charts
**/docker-compose*
**/compose*
**/Dockerfile*
**/node_modules
**/npm-debug.log
**/obj
**/secrets.dev.yaml
**/values.dev.yaml
README.md
```

**Docker build command**
```
$ docker build -t crawler .
```

**Docker run command**
```
$ docker run --env-file .env crawler
```

**Comment by SFaraji** [comment]
@vanekj any solution did you found? 
I'm having the same issue and hence opened a new request #90.

**Comment by SFaraji** [comment]
Ok, but can you now give a solution to the issue.

**Comment by SFaraji** [comment]
Hey, can someone please help regarding this ticket. 
@B4nan @vanekj

**Comment by vanekj** [comment]
Hi @SFaraji, I gave up on using the apify Dockerfile and using the Playwright one

```
# Get the base image of Node version 16
FROM node:16

# Get the latest version of Playwright
FROM mcr.microsoft.com/playwright:focal

# Set the work directory for the application
WORKDIR /app

# COPY the needed files to the app folder in Docker image
COPY package*.json /app/

# Get the needed libraries to run Playwright
RUN apt-get update && apt-get -y install libnss3 libatk-bridge2.0-0 libdrm-dev libxkbcommon-dev libgbm-dev libasound-dev libatspi2.0-0 libxshmfence-dev

# Install the dependencies in Node environment
RUN npm ci

# Start the main script
CMD ["node", "--inspect=0.0.0.0:9229", "src/main.js"]
```

**Comment by vanekj** [comment]
I am running it on my private VPS

**Comment by SFaraji** [comment]
Thanks @vanekj I really appreciate all fixed. Just wondering is there any other way to get the necessary libraries for Playwright? Because it increased my Docker image size.

**Comment by vanekj** [comment]
Unfortunately I did not play with it more to strip down the size as I was happy it's working for my needs.

**Comment by mnmkng** [comment]
AFAIK Apify packages are usually installed with `npm install` whereas you use `npm ci`. It might be the cause of your issues. Have you tried the [recommended Dockerfiles](https://sdk.apify.com/docs/guides/docker-images)?

**Comment by underfisk** [comment]
I'm also experiencing the same issue, looks like it might be broken 😅

**Comment by mnmkng** [comment]
We're running hundreds of thousands of runs and thousands of builds on those images every day, they're not broken per se. But they might be broken in some specific configurations. Please provide a reproduction scenario or more information. We would like to help, but without more info there's no way how.

**Comment by underfisk** [comment]
> We're running hundreds of thousands of runs and thousands of builds on those images every day, they're not broken per se. But they might be broken in some specific configurations. Please provide a reproduction scenario or more information. We would like to help, but without more info there's no way how.

I did post in another issue #91 where I'm using `pnpm` and running a Nestjs app with crawlee. Also a little bit more context is that I'm deploy to an ECS

**Comment by iBubelo** [comment]
I also have exactly the same issue.
@vanekj thanks for pointing to Playwright Docker image, it solved the issue for me

**Comment by kejiweixun** [comment]
Had the same issue.

In my case, it's caused by the mismatch of the playwright version between in package.json and in docker image.

For example, I follow the [crawlee doc](https://crawlee.dev/docs/guides/docker-images#best-practices), specify `*` for playwright in package.json, and get an image named `apify/actor-node-playwright-chrome:16-1.31.2`, this will cause the issue. If I replace `*` with `1.31.2`, it will be ok. 

In summary, this is my package.json:

```json
{
    "dependencies": {
       "@crawlee/playwright": "^3.3.0",
       "playwright": "1.31.2",
    }
}
```

This is my base image: `FROM apify/actor-node-playwright-chrome:16-1.31.2`

**Comment by kejiweixun** [comment]
@mnmkng Hi, I try to reproduce this issue, but couldn't. I changed lots of my code including the Dockerfile since I thought I "fixed" this issue, but I don't have a copy of my code when this issue occured.

Now in my code, I change `"playwright": "1.31.2"` to `"playwright": "*"`, and at the same time, I change `FROM apify/actor-node-playwright-chrome:16-1.31.2` to `FROM apify/actor-node-playwright-chrome:16`, it works with no problem.

**Comment by B4nan** [comment]
There are new releases of playwright (one landed just a few hours ago), so once we publish a new version of crawlee, the base docker image gets rebuilt and will contain a newer version - and I am afraid that will break it for you again. I feel like the pinning might be actually required to resolve this, as without it, you have two places that need to be synchronized but there is no link between them. NPM with `*` will technically try to resolve to the latest version, which might be not available in the docker file. If you have a lockfile, you generate that locally - so it will go for the latest available version, but that might not be available in the docker image yet, so fails on building it. And vice versa, if you have a lock file and rebuild an older project, you can get "too new docker image". I think that only with both dockerfile and package.json dependency pinned to exact version can resolve all the possible cases. On the other hand, this requires users to change two places when upgrading the playwright/puppeteer version.

**Comment by RowanAldean** [comment]
Writing for future reference. I was having issues specifically deploying to Cloud Run using `cloudbuild.yaml` and `gcloud builds submit` with expected steps to build, push and deploy a Crawlee project.

Further details of my error are in the [mentioned issue](https://github.com/apify/crawlee/issues/2275). I solved for this by adding `RUN npx playwright install` to my own `Dockerfile` thereby not depending on the base image to perform this install and ensuring all default browsers are present in the resulting image.

Good luck future Googler navigating this mess! 🫡

**Comment by jkorach** [comment]
Another option that worked for me is concating `&& npx playwright install` to the `posinstall` script in the `package.json` file
![image](https://github.com/apify/apify-actor-docker/assets/85932814/da472a54-781e-4181-a384-f69fe013aa4c)

**Comment by javaswing** [comment]
I had the same problem, but I didn't have it when I used a different version of the image

`FROM apify/actor-node-playwright-chrome:20-1.49.1 AS builder`

Good luck for you

**Comment by vladfrangu** [comment]
> So with that in mind, shouldn't the playwright version be pinned to something that is at least compatible with the dockerfile?

This is why docker images now get tagged with their compatible playwright version, so you can pin both. We might need to update our templates for it!

`FROM apify/actor-node-playwright-chrome:20-1.52.0 AS builder` and pinning it in package.json should ensure it will always (🤞) work
